packages/eslint-plugin/src/rules/space-before-blocks.ts

```diff
@@ -0,0 +1,102 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { getESLintCoreRule } from '../util/getESLintCoreRule';
+import * as util from '../util';
+
+const baseRule = getESLintCoreRule('space-before-blocks');
+
+export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'space-before-blocks',
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'Enforces consistent spacing before blocks',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: baseRule.meta.fixable,
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    schema: baseRule.meta.schema,
+    messages: {
+      // @ts-expect-error -- we report on this messageId so we need to ensure it's there in case ESLint changes in future
+      unexpectedSpace: 'Unexpected space before opening brace.',
+      // @ts-expect-error -- we report on this messageId so we need to ensure it's there in case ESLint changes in future
+      missingSpace: 'Missing space before opening brace.',
+      ...baseRule.meta.messages,
+    },
+  },
+  defaultOptions: ['always'],
+  create(context) {
+    const rules = baseRule.create(context);
+    const config = context.options[0];
+    const sourceCode = context.getSourceCode();
+
+    let alwaysClasses = true,
+      neverClasses = false;
```

Comment:
I found this and the always/never pattern a bit off but left it as because that's how it is in the original rule. Thought it would have been simpler to see differences. I'll update :)